tests/Integration/Behaviour/Features/Scenario/Product/update_seo.feature

```diff
@@ -0,0 +1,168 @@
+@reset-database-before-feature
+@clear-cache-before-feature
+@update-seo
+Feature: Update product SEO options from Back Office (BO)
+  As an employee
+  I need to be able to update product SEO options from Back Office
+
+  Scenario: I update product SEO
+    Given I add product "product1" with following information:
+      | name       | en-US: just boots         |
+      | is_virtual | false                     |
+    Given I add product "product2" with following information:
+      | name       | en-US: just boots         |
+      | is_virtual | false                     |
+    And product product2 should have following values:
+      | redirect_type    | 404                 |
+    And product product2 should not have a redirect target
+    And product "product2" localized "meta_title" is "en-US:"
+    And product "product2" localized "meta_description" is "en-US:"
+    And product "product2" localized "link_rewrite" is "en-US:"
+    When I update product product2 SEO information with following values:
+      | meta_title       | en-US:product2 meta title       |
+      | meta_description | en-US:product2 meta description |
+      | link_rewrite     | en-US:waterproof-boots          |
+      | redirect_type    | 301-product                     |
+      | redirect_target  | product1                        |
+    Then product "product2" localized "meta_title" should be "en-US:product2 meta title"
+    And product "product2" localized "meta_description" should be "en-US:product2 meta description"
+    And product "product2" localized "link_rewrite" should be "en-US:waterproof-boots"
+    And product product2 should have following values:
+      | redirect_type    | 301-product               |
+    And product product2 redirect target should be product1
+
+  Scenario: Update product redirect type without providing redirect target
+    Given product "product2" localized "meta_title" should be "en-US:product2 meta title"
+    And product "product2" localized "meta_description" should be "en-US:product2 meta description"
+    And product "product2" localized "link_rewrite" should be "en-US:waterproof-boots"
+    And product product2 should have following values:
+      | redirect_type    | 301-product               |
+    And product product2 redirect target should be product1
+    When I update product product2 SEO information with following values:
+      | redirect_type    | 302-product               |
+      | redirect_target  |                           |
+    Then I should get error that product redirect_target is invalid
+    And product "product2" localized "meta_title" should be "en-US:product2 meta title"
+    And product "product2" localized "meta_description" should be "en-US:product2 meta description"
+    And product "product2" localized "link_rewrite" should be "en-US:waterproof-boots"
+    And product product2 should have following values:
+      | redirect_type    | 301-product                |
```

Comment:
```suggestion
      | redirect_type    | 302-product                |
```
It would seems more logical to change this value since the redirect_type is already `301-product`

Reply:
I think you missed something. It is the outcome (the assertion), not the event. Im asserting that the value haven't changed after exception was thrown

Reply:
Oh right my bad 😅